api/src/main.js

```diff
@@ -44,5 +47,68 @@ if (import.meta.env.PROD) {
   });
 }
 
+// ---------------------------------------------------------------------------------------------------------------------
+// ---------------------------------------------------------------------------------------------------------------------
+// ---------------------------------------------------------------------------------------------------------------------
+
+const httpServer = createServer();
+const io = new Server(httpServer, {
+  cors: {
+    origin: "*",
+    credentials: true
+  }
+});
+const rooms = [];
+const users = new Map;
+
+
+
+app.post('/user', (req, res) => {
+  const data = req.body;
+  users.set("8f289a73-cf73-4931-ab93-84f65e3462b4",
+    {
+      email: "gotaga.gros+caca@gmail.com",
+      username: "ZroGizi",
+      password: "ZroGizi",
+      room: "random"
+    }
+  )
+  return res.status(200).json({
+    "uuid": "8f289a73-cf73-4931-ab93-84f65e3462b4",
+    "new_user": users.get("8f289a73-cf73-4931-ab93-84f65e3462b4")
+  })
+})
+
+app.get('/rooms', (req, res) => {
+  return res.status(200).json({
+    "rooms": rooms
+  })
+})
+
+function roomExist(givenRoomName) {
+  const isInRooms = rooms.find(room => room.name === givenRoomName)
+  return isInRooms !== undefined
+}
+
+function getRoomUserList(room) {
+  let roomUserList = []
+  users.forEach(user => {
+    if (room === user.room) {
+      roomUserList.push(user.username)
+    }
+  })
+  return roomUserList
+}
+
+const onConnection = (socket) => {
+  socketReceiveHandler(io, socket, users, roomExist, getRoomUserList, rooms);
+}
+
+io.on("connection", (socket) => {
+  onConnection(socket)
+})
+
+io.listen(3001);
```

Comment:
Use the same port as the HTTP server.

Here is an example: https://socket.io/get-started/chat/

Reply:
Make it work and then we talk :)

Reply:
We can talk now